### DISCUSSION TITLE
BeardLib Music Module: adding "allow_switch" per each track in the event

### DISCUSSION THREAD

**Discussion by MrVavlo** [diesel-modding/PAYDAY-2-BeardLib#633] [Ideas]
As of right now, `"allow_switch"` is working for either all the tracks used in the event, or doesn't allow switching to an alternative track until the end of the event.

I would like to see an option for something in-between: for example, allowing `track №1` to switch to `track №2` (or to other tracks, if there are any), but disable the switch when `track №2` was selected.

**Comment by segabl** [comment]
A very basic implementation that just lets you place the allow_switch flag in track notes should be simple enough to add

**Comment by segabl** [comment]
you can try this branch https://github.com/simon-wh/PAYDAY-2-BeardLib/tree/music-allow-switch

**Reply by segabl** [reply]
specifically you can now specify allow_switch in track nodes as well as the music node itself

**Reply by MrVavlo** [reply]
Alright, on my way to test it out!

**Reply by MrVavlo** [reply]
Tested - "false" flag is getting ignored and switches to another track.

Clarification: it's getting ignored if the track was switched in the event. It works if you get at track with the "false" flag from the start.

**Reply by segabl** [reply]
so it plays a track with switch allowed, switches to one that doesnt allow it but still switches afterwards?

**Reply by MrVavlo** [reply]
Correct.

**Reply by segabl** [reply]
ah yeah had an oversight, should work like intended now

**Reply by MrVavlo** [reply]
Can confirm, works smoothly. Thanks!
